dc_eval_models/deberta_base.py

```diff
@@ -0,0 +1,130 @@
+        return mean_embeddings
+
+class FeedBackModel(nn.Module):
+    def __init__(self, model_name):
+        super(FeedBackModel, self).__init__()
+        self.model = AutoModel.from_pretrained(model_name)
+        if CFG.gradient_checkpointing: (self.model).gradient_checkpointing_enable()
+        if CFG.freezing:
+            freeze((self.model).embeddings)
+            freeze((self.model).encoder.layer[:2])
+            CFG.after_freezed_parameters = filter(lambda parameter: parameter.requires_grad, (self.model).parameters())
+        self.config = AutoConfig.from_pretrained(model_name)
+        self.drop = nn.Dropout(p=CFG.dropout)
+        self.pooler = MeanPooling()
+        self.fc = nn.Linear(self.config.hidden_size, CFG.target_size)
+        
+    def forward(self, ids, mask):
+        out = self.model(input_ids = ids, attention_mask = mask, output_hidden_states = False)
+        out = self.pooler(out.last_hidden_state, mask)
+        out = self.drop(out)
+        outputs = self.fc(out)
+        return outputs.flatten()
+
+def replace_encoding_with_utf8(error: UnicodeError) -> Tuple[bytes, int]: return error.object[error.start : error.end].encode("utf-8"), error.end
+def replace_decoding_with_cp1252(error: UnicodeError) -> Tuple[str, int]: return error.object[error.start : error.end].decode("cp1252"), error.end
+codecs.register_error("replace_encoding_with_utf8", replace_encoding_with_utf8)
+codecs.register_error("replace_decoding_with_cp1252", replace_decoding_with_cp1252)
+
+def resolve_encodings_and_normalize(text: str) -> str:
+    text = (text.encode("raw_unicode_escape").decode("utf-8", errors = "replace_decoding_with_cp1252").encode("cp1252", errors = "replace_encoding_with_utf8").decode("utf-8", errors = "replace_decoding_with_cp1252"))
+    text = unidecode(text)
+    return text
+
+class DebertaBaseModel:
+    def __init__(self):
+        self.device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
+        self.target_list = ['Ineffective', 'Adequate', 'Effective']
+
+    def __prepare_input__(self, dc_text: str, dc_type: str, essay: str):
+        dc_text = resolve_encodings_and_normalize(dc_text)
+        essay = resolve_encodings_and_normalize(essay)
+        text = dc_type + ' ' + dc_text + '[SEP]' + essay
+
+        inputs = CFG.tokenizer.encode_plus(text, truncation=True, add_special_tokens=True, max_length=CFG.max_len)
+        return inputs
+
+    def predict(self, dc_text: str, dc_type: str, essay: str) -> dict:
+        data = self.__prepare_input__(dc_text, dc_type, essay)
+
+        ids = torch.tensor([data['input_ids']]).to(self.device, dtype = torch.long)
+        mask = torch.tensor([data['attention_mask']]).to(self.device, dtype = torch.long)
+
+        predictions = []
+        for i in CFG.trn_fold:
+            model = FeedBackModel(CFG.model)
+            model.load_state_dict(torch.load(f'./dc_eval_models/pretrained/deberta/fold_{i}.pth', map_location=self.device))
```

Comment:
Why on earth does this load the weights during inference? This would mean the weights are loaded all over again every time we call `predict(...)`, which then makes the inference speed ridiculously long.